onnxruntime/contrib_ops/cuda/custom_layer_norm.cu

```diff
@@ -0,0 +1,305 @@
+/* Modifications Copyright (c) Microsoft. */
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include <cooperative_groups.h>
+#include "custom_layer_norm.cuh"
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+namespace cg = cooperative_groups;
+using namespace onnxruntime::cuda;
+/*
+Custom kernel for Layer Normalization, adapted from https://github.com/microsoft/DeepSpeed.
```

Comment:
maybe refer to the file and commit too, e.g. https://github.com/microsoft/DeepSpeed/blob/4a3234e0ab3a6b0039741d7ea2fe0c1b82584763/csrc/transformer/normalize_kernels.cu